src/lib/api/org/index.ts

```diff
@@ -275,3 +277,108 @@ function convertToSnykProject(projectData: v3ProjectData[]) : SnykProject[] {
   return projects
 }
 
+export async function listTargets(
+  requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+): Promise<TargetsResponse> {
+  getApiToken();
+  getSnykHost();
+  debug(`Listing all targets for org: ${orgId}`);
+
+  if (!orgId) {
+    throw new Error(
+      `Missing required parameters. Please ensure you have set: orgId, settings.
+        \nFor more information see: https://apidocs.snyk.io/?version=2022-09-15%7Ebeta#get-/orgs/-org_id-/targets`,
+    );
+  }
+  
+  const targets = await listAllSnykTarget(requestManager, orgId, config)
+
+  return {targets: targets};
+}
+
+export async function listAllSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }
+  ): Promise<SnykTarget[]> {
+
+    let lastPage = false;
+    let SnykTargetList: SnykTarget[] = [];
+    let nextPageLink: string | undefined = undefined;
+    while (!lastPage) {
+      try {
+        const {
+          SnykTarget,
+          next,
+        }: {
+          SnykTarget: SnykTarget[];
+          next?: string;
+        } = await getSnykTarget(requestManager, orgId, nextPageLink, config);
+
+        SnykTargetList = SnykTargetList.concat(SnykTarget);
+        next
+          ? ((lastPage = false), (nextPageLink = next))
+          : ((lastPage = true), (nextPageLink = ''));
+      } catch (e) {
+        debug('Failed to get targets for ', orgId, e);
+        throw e;
+    }
+  }
+  return SnykTargetList
+}
+
+export async function getSnykTarget(requestManager: requestsManager,
+  orgId: string,
+  nextPageLink?: string, 
+  config?: {
+    limit?: number,
+    isEmpty?: boolean,
+    origin?: string,
+  }): Promise< { SnykTarget: SnykTarget[], next?: string } > {
+
+    const link = nextPageLink ? nextPageLink : `https://api.dev.snyk.io/rest/orgs/${orgId.trim()}/targets?version=2022-09-15~beta`
```

Comment:
I think @lili2311 mentioned something about hard coded URLs and that we should move to using the URL package.

Reply:
I had a look, it wouldn't help us at all

Reply:
I don't want to import a package for anything, the URL package is nice but mostly if you have parameters or if you need to specify the base, ....
the requests manager already does that.